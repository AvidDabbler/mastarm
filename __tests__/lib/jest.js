/* globals describe, it, expect */

const jestUtils = require('../../lib/jest')

describe('test.js', () => {
  it('generateTestConfig should generate proper config', () => {
    const cfg = jestUtils.generateTestConfig(['these', 'files', 'only'], {
      cache: false,
      coveragePaths: 'bin src another-folder',
      runInBand: true,
      setupFiles: 'beforeTestsSetup.js',
      testEnvironment: 'node',
      updateSnapshots: true
    })
    expect(cfg).toBeTruthy()
    expect(cfg.length).toEqual(8)
    const jestCfg = JSON.parse(cfg.splice(4, 1))
    expect(cfg).toMatchSnapshot()
    expect(jestCfg.transform['.*']).toContain('lib/jest-preprocessor.js')
    delete jestCfg.transform
    expect(jestCfg).toMatchSnapshot()
  })
})
