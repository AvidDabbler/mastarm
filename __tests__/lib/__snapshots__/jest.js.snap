// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test.js generateTestConfig should generate proper config 1`] = `
Array [
  "--forceExit",
  "--updateSnapshot",
  "--no-cache",
  "--runInBand",
  "--config",
  Object {
    "cacheDirectory": "tmp",
    "collectCoverageFrom": Array [
      "lib/**/*.js",
      "bin",
      "src",
      "another-folder",
    ],
    "coverageDirectory": "coverage",
    "notify": true,
    "setupFiles": Array [
      "beforeTestsSetup.js",
    ],
    "testEnvironment": "node",
    "testPathIgnorePatterns": Array [
      "<rootDir>/node_modules/",
      "<rootDir>/__tests__/test-utils",
    ],
  },
  "these",
  "files",
  "only",
]
`;
